specs/_features/eip7594/das-core.md

```diff
@@ -0,0 +1,225 @@
+
+## Peer scoring
+
+Due to the deterministic custody functions, a node knows exactly what a peer should be able to respond to. In the event that a peer does not respond to samples of their custodied rows/columns, a node may downscore or disconnect from a peer.
+
+## DAS providers
+
+A DAS provider is a consistently-available-for-DAS-queries, super-full (or high capacity) node. To the p2p, these look just like other nodes but with high advertised capacity, and they should generally be able to be latently found via normal discovery.
+
+DAS providers can also be found out-of-band and configured into a node to connect to directly and prioritize. Nodes can add some set of these to their local configuration for persistent connection to bolster their DAS quality of service.
+
+Such direct peering utilizes a feature supported out of the box today on all nodes and can complement (and reduce attackability and increase quality-of-service) alternative peer discovery mechanisms.
+
+## A note on fork choice
+
+*Fork choice spec TBD, but it will just be a replacement of `is_data_available()` call in Deneb with column sampling instead of full download. Note the `is_data_available(slot_N)` will likely do a `-1` follow distance so that you just need to check the availability of slot `N-1` for slot `N` (starting with the block proposer of `N`).*
+
+The fork choice rule (essentially a DA filter) is *orthogonal to a given DAS design*, other than the efficiency of a particular design impacting it.
+
+In any DAS design, there are probably a few degrees of freedom around timing, acceptability of short-term re-orgs, etc. 
+
+For example, the fork choice rule might require validators to do successful DAS on slot N to be able to include block of slot `N` in its fork choice. That's the tightest DA filter. But trailing filters are also probably acceptable, knowing that there might be some failures/short re-orgs but that they don't hurt the aggregate security. For example, the rule could be — DAS must be completed for slot N-1 for a child block in N to be included in the fork choice.
+
+Such trailing techniques and their analysis will be valuable for any DAS construction. The question is — can you relax how quickly you need to do DA and in the worst case not confirm unavailable data via attestations/finality, and what impact does it have on short-term re-orgs and fast confirmation rules.
+
+## FAQs
+
+### Row (blob) custody
+
+In the one-dimension construction, a node samples the peers by requesting the whole `DataColumn`. In reconstruction, a node can reconstruct all the blobs by 50% of the columns. Note that nodes can still download the row via `blob_sidecar_{subnet_id}` subnets.
```

Comment:
```suggestion
In the one-dimension construction, a node samples the peers by requesting the whole `DataColumn`. In reconstruction, a node can reconstruct all the blobs by 50% of the columns. Note that nodes can still download the row via `blob_sidecar_{subnet_id}` subnets from advertised supernodes.
```

Reply:
They can download these from any honest peers, as all honest peers have a deterministic custody requirement. 

Nodes that choose to custody beyond the baseline are a nice-to-have but are not requisite here

Reply:
It seems unclear to me whether we deprecate the `blob_sidecar_{subnet_id}` topics initially - I feel like the original intention in the ethresearch [post](https://ethresear.ch/t/from-4844-to-danksharding-a-path-to-scaling-ethereum-da/18046) to deprecate these topics was to not add extra bandwidth requirement on top of 4844 (prior to implementation of distributed reconstruction), while allowing an increase in max blob count.

I don't see `MAX_BLOBS_PER_BLOCK` increased here, feels like it make sense to increase this with PeerDAS? (otherwise we aren't really scaling)

However, if we increase `MAX_BLOBS_PER_BLOCK`, we are unlikely able to make regular nodes custody all rows / blbos, so we probably would need to either
1. Deprecate row subnets, and reconstruct blobs when required (expensive and potentially slow?)
2. Keep the row subnets, but make nodes custody a subset of all row subnets using a similar approach, e.g. `get_custody_row` .

Reply:
Sorry I commented pre-maturely - I've just read this: 

> However, for simplicity, we don't assign row custody assignments to nodes in the current design.

So does that mean each node custody _all_ rows and 1 column by default under the current proposal?

Reply:
> So does that mean each node custody _all_ rows and 1 column by default under the current proposal?

Each node custodies 1 subnet, which corresponds to 4 columns, and it does not custody any row.

Reply:
> Each node custodies 1 subnet, which corresponds to 4 columns, and it does not custody any row.
yes this is my understanding of it

Reply:
> Each node custodies 1 subnet, which corresponds to 4 columns, and it does not custody any row.

How do we easily serve Blobs to users of the Beacon endpoints? 

Without any nodes custodying row subnets, we'd need to retrieve 50% of the columns and reconstruct the full maxtrix right?

Reply:
I've just saw the DAS c-kzg benchmarks, the time for reconstruction isn't actually that bad, but wonder how quick we could get 50% of the samples.

Reply:
> Without any nodes custodying row subnets, we'd need to retrieve 50% of the columns and reconstruct the full maxtrix right?

Yes. Actually, there is no need to serve specific blobs.

Reply:
Deprecated `blob_sidecar_{subnet_id}
 ` in https://github.com/ethereum/consensus-specs/pull/3574/commits/d23452d02bb54e9bffc5547fde00cfd1537d6d42.

Feel free to reopen it if needed :)